File: go-app/internal/models/snapshot_item.go
package models

type SnapshotItem struct {
	Name           string  `json:"name"`
	PokemonCardID  string  `json:"pokemon_card_id,omitempty"`
	SetName        string  `json:"set_name,omitempty"`
	CardNumber     string  `json:"card_number,omitempty"`
	Rarity         string  `json:"rarity,omitempty"`
	Qty            int     `json:"qty"`
	MarketPrice    float64 `json:"market_price"`
	Percentage     float64 `json:"percentage"`
	Offered        float64 `json:"offered"`
	ItemType       string  `json:"item_type"` // RAW_CARD | GRADED_CARD | SEALED_PRODUCT
	IsTracked      bool    `json:"is_tracked"`
	Purpose        string  `json:"purpose,omitempty"` // PERSONAL_COLLECTION | PENDING_GRADE
	GradingCompany string  `json:"grading_company,omitempty"`
	Grade          string  `json:"grade,omitempty"`
	ImageURL       string  `json:"image_url,omitempty"`
}

File: src/main/java/com/collectingwithzak/dto/request/SnapshotItem.java
package com.collectingwithzak.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class SnapshotItem {

    private String name = "";

    @JsonProperty("pokemon_card_id")
    private String pokemonCardId = "";

    @JsonProperty("set_name")
    private String setName = "";

    @JsonProperty("card_number")
    private String cardNumber = "";

    private String rarity = "";

    private int qty = 1;

    @JsonProperty("market_price")
    private double marketPrice;

    private double percentage = 60;

    private double offered;

    @JsonProperty("item_type")
    private String itemType = "RAW_CARD";

    private String purpose = "";

    @JsonProperty("grading_company")
    private String gradingCompany = "";

    private String grade = "";

    @JsonProperty("is_tracked")
    private boolean isTracked;

    @JsonProperty("image_url")
    private String imageUrl = "";
}
